### DISCUSSION TITLE
Proposition : Add a module to display rich formatted text (`RTF`)

### DISCUSSION THREAD

**Discussion by d0rianb** [zed-industries/zed#11730] [Show and tell]
Would it be helpful to implement a module for displaying rich formatted text ? 
It could be a nice feature to be able to view and edit rtf documents in Zed without having to launch an external (often heavy) editor like Word.
Currently, rtf documents are displayed as raw content, which is not useful for most usage.
We could imagine a module like `markdown-preview` with both raw and formatted views side-by-side, or a preview in its own tab.
This module could be powered by the [`rtf-parser`](https://crates.io/crates/rtf-parser) crate, which is fast enough for real-time parsing.

I will start implementing this module on a fork of Zed to dig into GPUI and I wanted to know if this proposal would be interesting ?

**Comment by StuartMW-YL** [comment]
I'd love to see something for markdown first, and while I don't have much need for RTF, I can imagine others preferring this to markdown for their own reasons. I'm currently using Marked2 for viewing Markdown files.

**Reply by bennetbo** [reply]
@StuartMW-YL You can preview markdown files already by selecting `markdown: Open preview` inside the command palette (cmd-shift-p)

**Reply by d0rianb** [reply]
@StuartMW-YL As bennetbo said, there already is a `markdown-preview` package. 
The goal of this `rtf-preview` package is to mimic its functionnalities with richer text format, which could be used to visualise Word documents, for example.

**Comment by nahuel** [comment]
Related: https://github.com/zed-industries/zed/discussions/37270
